### PR TITLE
ci: try newer images

### PR DESCRIPTION
**Pull request by henryiii**
## Description

It seems MSVC 2022 is causing some issues (#3459, #3477). Let's try it explicitly.

## Suggested changelog entry:

```rst

```

### CODE REVIEW

**Comment by henryiii** [comment]
So it seems removing _DEBUG fixes the issue with needing the extra include (that is from an optional component), but reintroduces the original `LINK : fatal error LNK1104: cannot open file 'python35_d.lib'` type errors.

**Comment by snnn** [comment]
From my understanding, in that case you will need to used python_d.exe instead of python.exe to run the package. Then I will lost numpy because I don't know how to build numpy in such a way.

**Comment by henryiii** [comment]
Yes, this is why we disable _DEBUG when including Python.h unless this is a Py_DEBUG build. Python.h can't include C++ sources, though, so I don't know why MSVC 2022 is unhappy. I'm not sure adding a PYBIND11_KEEP_YOUR_HANDS_OFF_DEBUG variable would actually provide a way around the problem due to that.

**Comment by libbooze** [comment]
> Yes, this is why we disable _DEBUG when including Python.h unless this is a Py_DEBUG build. Python.h can't include C++ sources, though, so I don't know why MSVC 2022 is unhappy. 

Not a STL person but I read their comments as:
 your undefing of _DEBUG breaks their code because they think they can use a function that is not available in release version of their libs, so linker errors happen.
https://github.com/microsoft/STL/issues/2335#issuecomment-967311942

also they say _DEBUG is NOT a customization point so you are not allowed to do what you do (I learned from that issue that clang even has a [warning ](https://godbolt.org/z/cnE3jeaKf)when you mess with reserved identifiers).
https://github.com/microsoft/STL/issues/2335#issuecomment-967489428

tl;dr I don't think MS will ever agree to supporting this, if you explain underlying issue why originally you undef _DEBUG then maybe somebody reading this will be able to figure out a different hack/solution.

If something like that is not feasible then maybe just add to README that debug build is not supported.

**Comment by henryiii** [comment]
99% of users making debug builds want to debug _their_ code, not CPython itself; they usually don't have a CPython debug library and wouldn't know how to link to it even if they did. That's why we do this. If there's another way to get CPython to not require the debug library but to still build user code in debug mode, I'm all ears. :)

**Comment by libbooze** [comment]
> If there's another way to get CPython to not require the debug library but to still build user code in debug mode, I'm all ears. :)
Thank you for reply. I think best would be to ask CPython people if they would support this, maybe they can make their dependency in debug configurable. 

Beside that I think this hack might work:

https://github.com/pybind/pybind11/issues/3477#issuecomment-974899737
